### PR TITLE
Fix missing initializations and dead assignments in RecoMET/METAlgorithms

### PR DESCRIPTION
**Pull request by perrotta**
Found by static analyzer:
- Dead assignments in RecoMET/METAlgorithms/src/MuonMETAlgo.cc
- Missing initializations in RecoMET/METAlgorithms/src/GlobalHaloAlgo.cc

I wonder whether the missing initializations are anyhow affecting the final results (which should be the case, unless the compiler implicitly initialize to 0 all uninitialized integers...)

### CODE REVIEW

**Comment by perrotta** [comment]
@eioannou : this touches RecoMET

**Comment by perrotta** [comment]
please test

File: RecoMET/METAlgorithms/src/GlobalHaloAlgo.cc [hunk 1]
```diff
@@ -48,13 +48,18 @@ reco::GlobalHaloData GlobalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeomet
   float METOverSumEt = TheCaloMET.sumEt() ? TheCaloMET.pt() / TheCaloMET.sumEt() : 0 ;
   TheGlobalHaloData.SetMETOverSumEt(METOverSumEt);
 
-  //int EcalOverlapping_CSCRecHits[73];
-  //int EcalOverlapping_CSCSegments[73];
-
   int EcalOverlapping_CSCRecHits[361];
```
Comment (guitargeek) [inline RecoMET/METAlgorithms/src/GlobalHaloAlgo.cc]: Why not `int EcalOverlapping_CSCRecHits[361] = {};`? I think I have done something like that before, and it zero-initializes all elements.

Comment (perrotta) [reply RecoMET/METAlgorithms/src/GlobalHaloAlgo.cc]: Why not?
That's cleaner, thank you @guitargeek

**Comment by perrotta** [comment]
abort test

**Comment by perrotta** [comment]
please abort

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Trivial fixes
- Jenkins tests pass

**Pull request merged**